src/components/ChatEntry.js

```diff
@@ -1,22 +1,46 @@
-import React from 'react';
 import './ChatEntry.css';
 import PropTypes from 'prop-types';
+import TimeStamp from './TimeStamp';
 
 const ChatEntry = (props) => {
+  const onLikeButtonClick = () => {
+    const updateChatEntry = {
+      id: props.id,
+      sender: props.sender,
+      body: props.body,
+      timeStamp: props.timeStamp,
+      liked: !props.liked,
+    };
+    props.onUpdateLiked(updateChatEntry);
+  };
```

Comment:
While this approach is consistent with what's shown in Learn, it has the downside that a presentation component needs to have enough information to reconstruct an object (including data not required for display) and know how to mutate the new object in the process.

I prefer the approach where only the object id is passed to the event handler. Sending up an id allows the child component to stay simpler, as it only needs the id and any data required for actual display, leaving all the messy stuff to be handled by code that has more knowledge of what the full data might look like.

Plus, the only key-value pair we're updating is `liked`. If we're only changing one value in an object, we shouldn't have to recreate the object entirely! `data` in `app.js` already has that object in its entirety, we're doing extra work by recreating it in a child component. By passing the `id` we can simply search for the object in `data` and update the desired value.